core/src/main/scala/org/apache/spark/scheduler/MapStatus.scala

```diff
@@ -176,15 +179,17 @@ private[spark] object HighlyCompressedMapStatus {
     // From a compression standpoint, it shouldn't matter whether we track empty or non-empty
```

Comment:
This comment about compression is maybe generally slightly misleading. Even a BitSet can use a different amount of memory when you reverse the bits. Though some compression schemes (e.g., EWAH as used in Hive, see https://github.com/lemire/javaewah) are mostly symmetric with respect to bit flips, it is not the case with the RoaringBitmap library (though Lucene's Roaring implementation appears to be symmetric). I'd recommend either removing or qualifying this comment. (The comment is not wrong because it says "should".)